src/content/app/genome-browser/Browser.tsx

```diff
@@ -173,10 +188,107 @@ type GenomeBrowserContextType = {
   setCogList: (cogList: CogList) => void;
 };
 
+type GenomeBrowserIdsContextType = {
+  genomeIdInUrl: string | undefined;
+  genomeIdForUrl: string | undefined;
+  focusObjectIdInUrl: string | null;
+  isFetchingGenomeId: boolean;
+  isMissingGenomeId: boolean | undefined;
+  genomeId: string | undefined;
+  focusObjectId: string | undefined;
+  focusObjectIdForUrl: string | undefined;
+  parsedFocusObjectId:
+    | {
+        type: string;
+        objectId: string;
+        genomeId: string;
+      }
+    | undefined;
+  isMalformedFocusObjectId: boolean;
+};
+
 export const GenomeBrowserContext = React.createContext<
   GenomeBrowserContextType | undefined
 >(undefined);
 
+export const GenomeBrowserIdsContext = React.createContext<
```

Comment:
Let's move everything related to the GenomeBrowserIdsContext (the context provider, the types, the related imports and helper functions) into its own file. Maybe the `app/genome-browser` folder deserves a new `contexts` folder? We should move the genome browser context over there as well, in a separate PR.

Reply:
✅